### PR TITLE
Add custom rendering infrastructure

### PR DESCRIPTION
**Pull request by apfelbox**
* Move all classes to `src/` and all tests to `tests/`. That eases the integration with composer and phpunit.
* Add custom rendering twig function `route_tree_render(name, options)`
* Add custom simple route voter (that ignores the route parameters).
* Also add a custom template that uses the “navigation” translation domain by default.
* Set better default classes that more match our style

### CODE REVIEW

File: src/KnpMenu/Voter/SimpleRouteVoter.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+     * @param RequestStack $requestStack
+     */
+    public function __construct (RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+
+    /**
+     * @inheritDoc
```
Comment (keichinger) [inline src/KnpMenu/Voter/SimpleRouteVoter.php]: That method isn't inherited.

File: src/Resources/config/services.yaml [hunk 2]
```diff
@@ -14,3 +14,10 @@ services:
 
     Becklyn\RouteTreeBundle\PostProcessing\Processor\Security\SecurityChecker:
         $trustResolver: '@security.authentication.trust_resolver'
+
+    Becklyn\RouteTreeBundle\KnpMenu\Renderer\SimpleTwigRenderer:
+        tags:
+            - { name: knp_menu.renderer, alias: route_tree }
```
Comment (keichinger) [inline src/Resources/config/services.yaml]: Do we want to keep the duplicate hardcoding for the alias or do we want to use something like `!php/const:Becklyn\RouteTreeBundle\KnpMenu\Renderer\SimpleTwigRenderer::ALIAS`?

File: src/Resources/views/menu-template.html.twig [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{% extends 'knp_menu.html.twig' %}
+
+{% block label %}
+    {%- set translation_domain = item.extra('translation_domain', 'navigation') -%}
+    {%- set label = item.label -%}
+    {%- if translation_domain is not same as(false) -%}
+        {%- set label = label|trans(item.extra('translation_params', {}), translation_domain) -%}
+    {%- endif -%}
+    {%- if options.allow_safe_labels and item.extra('safe_label', false) %}{{ label|raw }}{% else %}{{ label }}{% endif -%}
```
Comment (keichinger) [inline src/Resources/views/menu-template.html.twig]: Only from reading the option names, shouldn't it be the other way around? Or is that option more or less to be interpreted as "did someone else sanitize the label? Yeah, let's dump it straight into the DOM".

File: src/KnpMenu/Voter/SimpleRouteVoter.php (hunk 1 above)
Comment (apfelbox) [reply src/KnpMenu/Voter/SimpleRouteVoter.php]: It is, but the interface was missing.

File: src/Resources/config/services.yaml (hunk 2 above)
Comment (apfelbox) [reply src/Resources/config/services.yaml]: Sure, why not.

btw: you are using the old syntax, the new one is: 

```yaml
key: !php/const Becklyn\RouteTreeBundle\KnpMenu\Renderer\SimpleTwigRenderer::ALIAS
```

File: src/Resources/views/menu-template.html.twig (hunk 3 above)
Comment (apfelbox) [reply src/Resources/views/menu-template.html.twig]: That is 99,9% the exactly same implementation as in the original bundle: https://github.com/KnpLabs/KnpMenuBundle/blob/master/src/Resources/views/menu.html.twig

The only thing changed is the default translation domain.

Comment (apfelbox) [reply src/Resources/views/menu-template.html.twig]: btw: the name is misleading, but it is intended to say "yeah, that label is already safe, you can paste it directly in the DOM".

**Comment by apfelbox** [comment]
We aren't using CS here, so i would not add it in this PR (but rather a separate PR).

And yeah, it's still not working in PHP 7.3

**Pull request merged**